lib/state_machine/types.hpp

```diff
@@ -18,9 +20,10 @@ struct TransitionKey {
 struct TransitionHasher {
   std::size_t operator()(const TransitionKey &key) const
```

Comment:
`TransitionKey` is a bad name; what does it contain? If there's no more meaningful name call it `SourceAndMessage`